packages/test/functional-tests/src/test/ddsHandleEncoding.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+	MockFluidDataStoreRuntime,
+	MockStorage,
+	MockDeltaConnection,
+	MockHandle,
+} from "@fluidframework/test-runtime-utils";
+import { DirectoryFactory, IDirectory } from "@fluidframework/map";
+import { IChannel, IChannelFactory } from "@fluidframework/datastore-definitions";
+import { ConsensusRegisterCollectionFactory } from "@fluidframework/register-collection";
+
+/**
+ * The purpose of these tests is to demonstrate that DDSes do not do opaque encoding of handles
+ * when preparing the op payload (e.g. prematurely serializing).
+ * This is important because the runtime needs to inspect the full op payload for handles.
+ */
+describe("DDS Handle Encoding", () => {
+	const handle = new MockHandle("whatever");
+	const messages: any[] = [];
+
+	beforeEach(() => {
+		messages.length = 0;
+	});
+
+	/**
+	 * The purpose of the tests using this helper is to ensure that the message contents submitted
+	 * by the DDS contains handles that are detectable via JSON.stringify replacers, since this
+	 * is the technique used in the ContainerRuntime to detect handles in incoming ops.
```

Comment:
> since this is the technique used in the ContainerRuntime to detect handles in incoming ops.

I understand what you mean by this. But, the way container runtime detects handles is by parsing the op that has this serialized content. So, we should also add that validation, i.e., whatever logic container runtime uses to detect handles, the tests should use that and validate that the handles are the same as detected during this serialization.

Reply:
Yeah, now that it's a free function in dataStores.ts I'm going to have it just use the same function.  Yay.